pkg/config/config_new.go

```diff
@@ -0,0 +1,161 @@
+// config/config.go
+package config
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var DefaultConfigHandler = New()
+
+// ConfigHandler holds the application's configuration.
+type ConfigHandler struct {
+	atmosConfig *schema.AtmosConfiguration
+	v           *viper.Viper
+}
+
```

Comment:
Use interfaces makes testing easier by allowing mock implementations
```suggestion
 //ConfigHandler interface defines the contract for configuration handling
type ConfigHandler interface {
    AddConfig(cmd *cobra.Command, opts *ConfigOptions)
    Get() *schema.AtmosConfiguration
    BindEnv(key ...string)
    GetString(key string) string
    GetInt(key string) int
    GetBool(key string) bool
    SetDefault(key string, value any)
    Load(configAndStacksInfo *schema.ConfigAndStacksInfo) error
}
// configImpl holds the application's configuration implementation
type configImpl struct {
    atmosConfig *schema.AtmosConfiguration
    v           *viper.Viper
}
// New creates a new ConfigHandler instance with initialized Viper.
func New() ConfigHandler {
    v := viper.New()
    v.SetConfigType("yaml")
    v.SetTypeByDefaultValue(true)
    return &configImpl{
        v:           v,
        atmosConfig: &schema.AtmosConfiguration{},
    }
}
```